src/gui/uithememanager.cpp

```diff
@@ -58,6 +61,88 @@ namespace
 
         return {};
     }
+
+    QByteArray readFile(const QString &fileName)
+    {
+        QFile file {fileName};
+        if (!file.open(QIODevice::ReadOnly | QIODevice::Text))
+        {
+            LogMsg(UIThemeManager::tr("UITheme - Failed to open \"%1\". Reason: %2")
+                    .arg(QFileInfo(fileName).fileName(), file.errorString())
+                   , Log::WARNING);
+            return {};
+        }
+
+        return file.readAll();
+    }
+
+    class QRCThemeSource final : public UIThemeSource
+    {
+    public:
+        QByteArray styleSheet() override
+        {
+            return readFile(QRC_THEME_DIR + STYLESHEET_FILE_NAME);
+        }
+
+        QByteArray config() override
+        {
+            return readFile(QRC_THEME_DIR + CONFIG_FILE_NAME);
+        }
+
+        QString iconPath(const QString &iconId) override
+        {
+            return findIcon(iconId, QRC_ICONS_DIR);
+        }
+
+    private:
+        /* static */ const QString QRC_THEME_DIR {":/uitheme/"};
+        /* static */ const QString QRC_ICONS_DIR = QRC_THEME_DIR + THEME_ICONS_DIR;
+    };
+
+    class FolderThemeSource final : public UIThemeSource
+    {
+    public:
+        explicit FolderThemeSource(const QDir &dir)
+            : m_folder {dir}
+            , m_iconsDir {m_folder.absolutePath() + "/icons/"}
+        {
+        }
+
+        QByteArray styleSheet() override
+        {
+            QByteArray styleSheetData = readFile(m_folder.absoluteFilePath(STYLESHEET_FILE_NAME));
+            return styleSheetData.replace(STYLESHEET_ICONS_DIR, (m_folder.absolutePath() + '/').toUtf8());
+        }
+
+        QByteArray config() override
+        {
+            return readFile(m_folder.absoluteFilePath(CONFIG_FILE_NAME));
+        }
+
+        QString iconPath(const QString &iconId) override
+        {
+            return findIcon(iconId, m_iconsDir);
+        }
+
+    private:
+        const QDir m_folder;
+        const QString m_iconsDir;
+    };
+
+
+    std::unique_ptr<UIThemeSource> createUIThemeSource(const QString &themePath)
+    {
+        const QFileInfo themeInfo {themePath};
+
+        if (themeInfo.fileName() == CONFIG_FILE_NAME)
+            return std::make_unique<FolderThemeSource>(themeInfo.dir().absolutePath());
+
+        if ((themeInfo.suffix() == "qbtheme")
```

Comment:
```suggestion
        if ((themeInfo.suffix() == QLatin1String("qbtheme"))
```

Reply:
I thought you guys love affair with QLatin1String ended with qt6.

Reply:
> I thought you guys love affair with QLatin1String ended with qt6.

???

Reply:
With the source code encoding being utf8, QLatin1String makes a lot less sense https://lists.qt-project.org/pipermail/development/2020-May/039455.html

Reply:
> With the source code encoding being utf8, QLatin1String makes a lot less sense

Really? It doesn't look like anything has changed much in Qt6 since Qt5. Parsing of UTF-8 strings is slower than one of so-called Latin-1 strings (even if the UTF-8 string actually contains only Latin-1 characters, as in this case, the parser does not know about it, so it still has to check every character to understand what kind is it of). Am I wrong?

Reply:
hmm I guess it's a little performance improvement in non performance-critical code vs obscure casting everywhere

Reply:
Moreover comparison with QLatin1String doesn't even produces QString instance from it.

Reply:
AFAIK there is a `QString::operator =(const char*)`, so it shouldn't create a QString in this case either.

Reply:
> AFAIK there is a `QString::operator =(const char*)`, so it shouldn't create a QString in this case either.

I suppose you mean `bool	operator==(const char *other) const`. You could read the documentation:
>This function overloads `operator==()`.
>
The other const char pointer is converted to a `QString` using the `fromUtf8()` function.